gdx/src/com/badlogic/gdx/scenes/scene2d/ui/ScrollPane.java

```diff
@@ -173,7 +184,9 @@ public boolean mouseMoved (InputEvent event, float x, float y) {
 				return false;
 			}
 		});
+	}
 
+	protected void addFlickScrollListener() {
 		flickScrollListener = new ActorGestureListener() {

```

Comment:
Overriding this with a no-implementation is not possible, will give NPE.

So this should be a method returning a listener instead, annotated with `NonNull`. Ah wait, we don't have a `NonNull` annotation and don't need it because in libgdx returning null is not allowed anyway.

Reply:
Oh, good catch! I did the editing in the GitHub text editor a bit too hastily, apparently.

But now you make me wonder about my approach to allow `addListeners()` to be overridable: if `addFlickScrollListener()` returned a Listener to try to reduce the possibility that a user ends up with an NPE, what about the possibility that the user inadvertently overrides `addListeners()` and forgets to assign `flickScrollListener ` ?

Tell me what you think: I'd do what you suggested, and also change `addListeners()`'s visibility to `private`.

Reply:
Oh, good catch! I did the editing in the GitHub text editor a bit too hastily, apparently.

But now you make me wonder about my approach to allow `addListeners()` to be overridable: if `addFlickScrollListener()` returned a Listener to try to reduce the possibility that a user ends up with an NPE, what about the possibility that the user inadvertently overrides `addListeners()` and forgets to assign `flickScrollListener ` ?

Tell me what you think: I'd do what you suggested, and also change `addListeners()`'s visibility to `private`.